From 7ad549b89dfbfb67fd1cbbf6cafdd996e4722246 Mon Sep 17 00:00:00 2001
From: "Owen W. Taylor" <otaylor@fishsoup.net>
Date: Wed, 9 Sep 2020 16:14:57 -0400
Subject: [PATCH] OCI: extract appstream data for runtimes

Runtimes also have appstream data - with description, license information,
and so forth, so we should extract the appstream data from the index
for refs that start with runtime/ as well.
---
 common/flatpak-oci-registry.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/flatpak-oci-registry.c b/common/flatpak-oci-registry.c
index cf501a5eb..0f7471129 100644
--- a/common/flatpak-oci-registry.c
+++ b/common/flatpak-oci-registry.c
@@ -3120,7 +3120,8 @@ add_image_to_appstream (SoupSession               *soup_session,
     return;
 
   ref_parts = g_strsplit (ref, "/", -1);
-  if (g_strv_length (ref_parts) != 4 || strcmp (ref_parts[0], "app") != 0)
+  if (g_strv_length (ref_parts) != 4 ||
+      (strcmp (ref_parts[0], "app") != 0 && strcmp (ref_parts[0], "runtime") != 0))
     return;
 
   id = ref_parts[1];
